### PR TITLE
Move CMP CLI test output files to $BLDTOP/test-runs/test_cmp_cli/

### PR DESCRIPTION
**Pull request by DDvO**
This is the real fix for the issue addressed by #12925.

- [x] tests are added or updated

### CODE REVIEW

File: test/recipes/81-test_cmp_cli.t
Comment (levitte) [inline test/recipes/81-test_cmp_cli.t:206]: I would start by replacing this line with `subtest "name-of-subtest" => sub {`

Yes, it probably means that you will have to use `data_dir` and `data_file` to reach the source input files where you hadn't expected to have to.  However, you don't need to do much with your output files, just write them in the location this test runs in.

Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t:206]: The CMP CLI-based tests use much greater variety of input files than output files.
Therefore it was less effort to prepend the output files with BLDTOP than it would be the other way round.
And a stronger point against reverting this to prepending the input files is that they dynamically depend on the test server used (the Mock server is just one of them; at Siemens we use a couple of others, too) while on the other hand the main test configuration file `test.cnf`, which references several server-dependent input files by relative file names, is static.

Comment (levitte) [reply test/recipes/81-test_cmp_cli.t:206]: In that case, I would like to see a comment to that effect, as this isn't "normal" behaviour for our test recipes.

Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t:215]: I've added a comment giving essentially the explanation that I wrote above.

**Comment by DDvO** [comment]
@levitte, I've addressed your comments and all tests pass.
Anything else to do, or can this be approved now?

File: test/recipes/81-test_cmp_cli.t
Comment (levitte) [inline test/recipes/81-test_cmp_cli.t:73]: Actually, there is a supported command in OpenSSL::Test: `result_file`.  So you could change this line to:

``` perl
    my $outfile = result_file('test.certout.pem');
```

That way, you don't have to keep track of the exact directory yourself (i.e. you can ditch `$tmpdir`)

File: test/recipes/81-test_cmp_cli.t
Comment (levitte) [inline test/recipes/81-test_cmp_cli.t:260]: In the same vein, maybe there should be a `result_dir` in OpenSSL::Test as well...

File: test/recipes/81-test_cmp_cli.t
Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t]: Nice. I've change the test code that way,
after adapting the name of `__results_file` to `result_file` in OpenSSL::Test and exporting it.

File: test/recipes/81-test_cmp_cli.t
Comment (DDvO) [reply test/recipes/81-test_cmp_cli.t]: Done, after introducing `result_dir` in OpenSSL::Test.

File: util/perl/OpenSSL/Test.pm
Comment (levitte) [inline util/perl/OpenSSL/Test.pm:1069]: You dropped this line.   Don't.

Comment (DDvO) [reply util/perl/OpenSSL/Test.pm:1069]: I did so deliberately because this check is anyway done by `result_dir()` called by this function.
Isn't that sufficient?

Comment (DDvO) [reply util/perl/OpenSSL/Test.pm:1071]: All right, anyway re-add that BAIL_OUT.

**Comment by DDvO** [comment]
Ping for approval

**Comment by DDvO** [comment]
@levitte, can you approve this now?

**Comment by DDvO** [comment]
@levitte, also this PR is waiting for further handling/approval since nearly two weeks meanwhile..

**Review by t8m — APPROVED** [review]
LGTM

**Comment by levitte** [comment]
> @levitte, also this PR is waiting for further handling/approval since nearly two weeks meanwhile..

Sorry, I thought I had already done that...  too many other distractions, I guess

**Comment by levitte** [comment]
Merge it

**Comment by DDvO** [comment]
Merged - thanks @levitte for your comments and @t8m for the approval.

**Pull request closed without merging**
